distributed/worker.py

```diff
@@ -1707,17 +1714,11 @@ async def batched_send_connect():
     async def get_data(
         self,
         comm: Comm,
-        keys: Collection[str] | None = None,
+        keys: Collection[str],
         who: str | None = None,
         serializers: list[str] | None = None,
-        max_connections: int | None = None,
     ) -> GetDataBusy | Literal[Status.dont_reply]:
-        if max_connections is None:
-            max_connections = self.transfer_outgoing_count_limit
-
-        if keys is None:
-            keys = set()
-
+        max_connections = self.transfer_outgoing_count_limit
         # Allow same-host connections more liberally
         if (
             max_connections
```

Comment:
nit: IIUC, max_connections is now always an integer, so this can be dropped. The same goes for line 1738/1739 `max_connections is not False`.